src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -1314,6 +1320,16 @@ def forward(self, inputs, targets):
             if not self.ddp_silent_mode:
                 self.sg_logger.close()
 
+    def _set_net_preprocessing_from_valid_loader(self):
+        if isinstance(self.net.module, HasPredict) and isinstance(self.valid_loader.dataset, HasPreprocessingParams):
+            try:
+                self.net.module.set_dataset_processing_params(**self.valid_loader.dataset.get_dataset_preprocessing_params())
+            except Exception as e:
+                logger.warning(
+                    f"Could not set preprocessing pipeline from the validation dataset: {e}. Before calling"
+                    "predict make sure to call set_dataset_processing_params."
```

Comment:
Maybe adding "\n" before "Before calling [...]" would help because otherwise it would be in the same line as the error (so ppl probably won't see it)

Reply:
Added in latest commit.